### DISCUSSION TITLE
MFIA, How do I get the Sweep values from the Web App GUI into Python?

### DISCUSSION THREAD

**Discussion by woopelderly** [zhinst/zhinst-toolkit#288] [Q&A] [answered]
When I hover over things in the web app GUI it shows their node path.  Most/all of the values in the upper portion of the GUI are accurately read using the node path with the zhinst-toolkit as expected.

The trouble I'm having is getting the values that have the node path "sweep/start", or "sweep/stop", etc.  Pictured is the sweep/start path.
![image](https://github.com/user-attachments/assets/c5876f87-1825-4d29-aa4a-b5f9346e3005)

In my Python application, my sweeper object does not have values that match the GUI Web App.  Is there a way to get those values from the GUI?  Or do I need to get/create my sweeper object differently?

Here is what I currently do to get my sweeper object:

```python
    from zhinst.toolkit import Session
    session = Session("valid ip")
    device = session.connect_device("valid device")

    sweeper = session.modules.sweeper
    sweeper.device(device)
    sweeper.gridnode(device.oscs[0].freq)

    print(f"{sweeper.start()=}") # value printed here does not match the web app.
```

**Comment by ghost** [comment, marked as answer]
The LabOne modules live directly within the client. This prevents them from being shared with other API instances. 

If you want to recreate the module (with all its configurations), the easiest way is to use the LabOne UI's API logging feature.

https://docs.zhinst.com/labone_api_user_manual/description_and_guidelines/how_to_labone_ui_log.html?h=api+log#how-to-use-the-labone-ui-logging-feature

(There is also the option to generate zhinst-toolkit code directly, even though the documentation is still missing this info)

**Reply by woopelderly** [reply]
That makes sense why I see somethings update in my python code while other things do not. Thank you. The command log looks very helpful!
